src/lobpcg/svd.rs

```diff
@@ -0,0 +1,327 @@
+use super::random;
+///! Truncated singular value decomposition
+///!
+///! This module computes the k largest/smallest singular values/vectors for a dense matrix.
+use crate::lobpcg::{lobpcg, LobpcgResult, TruncatedOrder as Order};
+use crate::Result;
+use ndarray::prelude::*;
+use num_traits::{Float, NumCast};
+use std::iter::Sum;
+use std::ops::DivAssign;
+
+/// The result of a eigenvalue decomposition, not yet transformed into singular values/vectors
+///
+/// Provides methods for either calculating just the singular values with reduced cost or the
+/// vectors with additional cost of matrix multiplication.
+#[derive(Debug)]
+pub struct TruncatedSvdResult<A> {
+    eigvals: Array1<A>,
+    eigvecs: Array2<A>,
+    problem: Array2<A>,
+    ngm: bool,
+}
+
+impl<A: Float + PartialOrd + DivAssign<A> + 'static + MagnitudeCorrection> TruncatedSvdResult<A> {
+    /// Returns singular values ordered by magnitude with indices.
+    fn singular_values_with_indices(&self) -> (Array1<A>, Vec<usize>) {
```

Comment:
Is it possible to order the singular value ascending instead of descending?

Reply:
the order is now defined by the order of the decomposition